sim/rogue/items_sets_pve.go

```diff
@@ -213,3 +215,132 @@ var ItemSetNightSlayerBattlearmor = core.NewItemSet(core.ItemSet{
+
+			aura := rogue.RegisterAura(core.Aura{
+				Label:    "Clearcasting (S03 - Item - T2 - Rogue - Damage 2P Bonus)",
+				ActionID: core.ActionID{SpellID: 467735},
+				Duration: time.Second * 15,
+				OnInit: func(aura *core.Aura, sim *core.Simulation) {
+					affectedSpells = core.FilterSlice(
+						rogue.Spellbook,
+						func(spell *core.Spell) bool {
+							return spell != nil && spell.Cost != nil && spell.Cost.CostType() == core.CostTypeEnergy
+						},
+					)
+				},
+				OnGain: func(aura *core.Aura, sim *core.Simulation) {
+					core.Each(affectedSpells, func(spell *core.Spell) { spell.Cost.Multiplier -= 100 })
+				},
+				OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+					core.Each(affectedSpells, func(spell *core.Spell) { spell.Cost.Multiplier += 100 })
+				},
+				OnCastComplete: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell) {
+					if aura.RemainingDuration(sim) == aura.Duration || spell.DefaultCast.Cost == 0 {
+						return
+					}
+					aura.Deactivate(sim)
+				},
+			})
+
+			core.MakePermanent(rogue.RegisterAura(core.Aura{
+				Label:    "S03 - Item - T2 - Rogue - Damage 2P Bonus",
+				OnSpellHitDealt: func(_ *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+					if !result.Landed() || !spell.ProcMask.Matches(core.ProcMaskMeleeOrRangedSpecial) {
+						return
+					}
+					if spell.SpellCode == SpellCode_RogueAmbush || spell.SpellCode == SpellCode_RogueGarrote {
+						aura.Activate(sim)
+					}
+				},
+			}))
+		},
+		// Increases main hand weapon damage by 10%
+		4: func(agent core.Agent) {
+			rogue := agent.(RogueAgent).GetRogue()
+			rogue.OnSpellRegistered(func(spell *core.Spell) {
+				if spell.ProcMask.Matches(core.ProcMaskMeleeMH) {
+					spell.DamageMultiplier *= 1.10
+				}
+			})
+		},
+		// Reduces cooldown on vanish by 2.5 minutes
+		6: func(agent core.Agent) {
+			rogue := agent.(RogueAgent).GetRogue()
+			rogue.RegisterAura(core.Aura{
+ 				Label: "S03 - Item - T2 - Rogue - Damage 6P Bonus",
+  				OnInit: func(aura *core.Aura, sim *core.Simulation) {
+    				rogue.Vanish.CD.Duration -= time.Second * 150
+  				},
+			})
+		},
+	},
+})
+
+var ItemSetBloodfangBattlearmor = core.NewItemSet(core.ItemSet{
+	Name: "Bloodfang Battlearmor",
+	Bonuses: map[int32]core.ApplyEffect{
+		// Your Rolling with the Punches now also activates every time you gain a combo point.
+		2: func(agent core.Agent) {
+			rogue := agent.(RogueAgent).GetRogue()
+			if !rogue.HasRune(proto.RogueRune_RuneRollingWithThePunches) {
+				return
+			}
+
+			rogue.OnComboPointsGained(func(sim *core.Simulation) {
+				if rogue.RollingWithThePunchesProcAura.GetStacks() == 5 {
+					rogue.RollingWithThePunchesProcAura.Refresh(sim)
+				} else if rogue.RollingWithThePunchesProcAura.IsActive() {
+					rogue.RollingWithThePunchesProcAura.AddStack(sim)
+				} else {
+					rogue.RollingWithThePunchesProcAura.Activate(sim)
+					rogue.RollingWithThePunchesProcAura.AddStack(sim)
+				}
```

Comment:
I _think_ that if you just Activate and add a stack every time here it should work the same way?

Reply:
Just attempted both way and read through the aura.go section on activate it does not seem to set stacks.  In testing this also flows as I lose the first proc without adding a stack.

Reply:
Cleaned up